apps/sms/services/js/moz_mobile_connections/moz_mobile_connections_shim.js

```diff
@@ -0,0 +1,104 @@
+/* global BridgeServiceMixin, BroadcastChannel, MozSettingsShim */
+
+/* exported MozMobileConnectionsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozMobileConnections API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-mobile-connections-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['getServiceIdByIccId', 'switchMmsSimHandler']);
+
+var mozMobileConnections = null;
+
+function setMmsSimServiceId(id) {
+  // DSDS: mms & data are both necessary for connection switch.
+  MozSettingsShim.set({
+    'ril.mms.defaultServiceId': id,
+    'ril.data.defaultServiceId': id
+  });
+}
+
+var MozMobileConnectionsShim = {
+  init(conns) {
+    if (!conns) {
+      console.error('Invalid settings for shim initialization');
+      return;
+    }
+    mozMobileConnections = conns;
+
+    this.initService(
+      new BroadcastChannel('moz-mobile-connections-shim-channel')
```

Comment:
> I was trying to listen on iframe originally, but it seems not working if I set the endpoint to iframe. I think the service side we don't need to set the endpoint for window <-> iframe communication right?

Yeah, for service it's just ```initService```, for client it's just ```client({....endpoint: iframe.....})```. If it doesn't work, it should be a bug though.